app/scripts/lib/stream-utils.js

```diff
@@ -50,3 +50,52 @@ export function isStreamWritable(stream) {
     stream.writable && !stream.destroyed && !stream._writableState?.ended,
   );
 }
+
+/**
+ * Creates a duplex stream wrapper for a substream.
+ *
+ * @param {ObjectMultiplex} mx - The multiplexer.
+ * @param {string} name - The name of the substream to create.
+ * @returns {Duplex} A fresh duplex that simply forwards data to/from substream.
+ */
+export function createDuplexStreamWrapper(mx, name) {
+  const substream = mx.createStream(name);
+
+  const duplexWrapper = new Duplex({
+    objectMode: true,
+    read() {
+      // We do nothing special here; data is pushed when the internal substream emits 'data'
+    },
+    write(chunk, _encoding, callback) {
+      try {
+        substream.write(
+          {
+            name: chunk.name,
+            ...chunk.data,
+          },
+          callback,
+        );
+      } catch (error) {
+        // eslint-disable-next-line node/callback-return
+        callback(error); // No return needed as we're catching synchronous errors
+      }
+    },
+  });
+
+  substream.on('data', (chunk) => {
+    duplexWrapper.push({
+      name,
+      data: chunk,
+    });
+  });
+
+  substream.on('error', (error) => {
```

Comment:
Hmm, is this what we want to do? That seems dangerous.

Reply:
yes make sense I just did drop it